platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+/**
+ * # PlatformState
+ * Messages that hold platform state in the network state.
+ *
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119)
+ * and clarified in [RFC8174](https://www.ietf.org/rfc/rfc8174).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.state;
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.state.legacy";
```

Comment:
Why is this in a "legacy" package? Isn't this intended to be the long term solution going forward?

Reply:
This is needed to work around a [PBJ bug](https://github.com/hashgraph/pbj/issues/240) (and [related](https://github.com/hashgraph/pbj/issues/218)).
Nothing _should_ actually be using this package value (PBJ doesn't generate classes with this package, and platform protos aren't generated with protoc).
Ideally we wouldn't set `java_package` at all (the main `package` statement should be what we intend for Java), but we need to work within the existing bugs at the moment.